lib/public/components/common/form/inputs/DateTimeInputModel.js

```diff
@@ -0,0 +1,316 @@
+
+/**
+ * @typedef DateTimeInputRawData
+ * @property {string} date the raw date value
+ * @property {string} time the raw time value
+ */
+
+/**
+ * @typedef DateTimeInputConfiguration
+ * @property {Partial<DateTimeInputRawData>} [defaults] the default raw values to use when partially updating inputs
+ * @property {boolean} [required] states if the input can be null
+ * @property {number} [min] the timestamp of the minimum date allowed by the input (included)
+ * @property {number} [max] the timestamp of the maximum date allowed by the input (excluded)
+ */
+
+/**
+ * Store the state of a date time model
+ */
+export class DateTimeInputModel extends Observable {
+    /**
+     * Constructor
+     * @param {DateTimeInputConfiguration} [configuration] the input configuration
+     */
+    constructor(configuration) {
+        super();
+        const { defaults = {}, required = false, min = null, max = null } = configuration;
+
+        /**
+         * @type {DateTimeInputRawData}
+         * @private
+         */
+        this._raw = { date: '', time: '' };
+
+        const { date: defaultDate = '', time: defaultTime = '' } = defaults;
+
+        /**
+         * @type {DateTimeInputRawData}
+         * @private
+         */
+        this._defaults = {
+            date: defaultDate,
+            time: defaultTime,
+        };
+
+        /**
+         * @type {number|undefined}
+         * @private
+         */
+        this._value = undefined;
+
+        this._visualChange$ = new Observable();
+
+        this._isRequired = required;
+        this._min = min;
+        this._max = max;
+    }
+
+    /**
+     * Returns the given date formatted in two parts, YYYY-MM-DD and HH:MM to fill in HTML date and time inputs
+     *
+     * @param {number} timestamp the timestamp (ms) to format
+     *
+     * @return {{date: string, time: string}} the date expression to use as HTML input values
+     */
+    formatDateForHTMLInput(timestamp) {
+        // Apply offset to have the inputs displayed in the user's timezone
+        return formatUTCDateForHTMLInput(timestamp - new Date(timestamp).getTimezoneOffset() * 60 * 1000);
+    }
+
+    /**
+     * Convert the text from HTML date and time input to a timestamp
+     *
+     * @param {string} date the date input's value
+     * @param {string} time the time input's value
+     * @return {number} the resulting timestamp
+     */
+    extractTimestampFromHTMLInput(date, time) {
+        // Apply offset because the inputs are displayed in the user's timezone
+        const inputsTimestamp = extractUTCTimestampFromHTMLInput(date, time);
+        return inputsTimestamp + new Date(inputsTimestamp).getTimezoneOffset() * 60 * 1000;
+    }
+
+    /**
+     * Update the inputs raw values
+     *
+     * @param {Partial<DateTimeInputRawData>} patch the input values patch
+     * @return {void}
+     */
+    update({ date, time }) {
+        if (date !== undefined) {
+            this._raw.date = date;
```

Comment:
what if null

Reply:
It can't be null (`Partial<DateTimeInputRawData>`)

Reply:
ok